### PR TITLE
Give examples of test-driven that worked

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

File: lifecycle.md [hunk 1]
```diff
@@ -46,13 +46,14 @@ For a mature standard, the whole process can repeat for proposed changes. New id
 
 As the standard, test suite and implementations co-evolve, an increasing number of tests will pass everywhere. For those tests, however trivial, full interop has been achieved. Once achieved, tooling should make it impossible to accidentally depart from full interop. Web developers can depend on the feature and delight their users, in any browser.
 
-## What about?
+## Examples
+ * DOM: mature [standard](https://dom.spec.whatwg.org/) and [test suite](https://github.com/w3c/web-platform-tests/tree/master/dom).
+ * HTML: the messagerror event had [spec text](https://github.com/whatwg/html/pull/2530) and [tests](https://github.com/w3c/web-platform-tests/pull/5567) written before any implementation.
```
Comment (annevk) [inline lifecycle.md]: And critically before the change to the standard landed.

File: lifecycle.md [hunk 2]
```diff
@@ -46,13 +46,14 @@ For a mature standard, the whole process can repeat for proposed changes. New id
...
+ * Trusted Types: [explainer](https://github.com/mikewest/trusted-types/blob/master/README.md) and [tentative tests](https://github.com/w3c/web-platform-tests/tree/cbc2da38b90e0870ac50a205d2fc2773de41bd5d/trusted-types).
 
-### Writing tests even earlier?
+*Note: Whether writing tests first or implementing first is most productive depends on circumstances. The most important thing is that tests are written and shared, not the order.*
```
Comment (annevk) [inline lifecycle.md]: I think we do have a quality problem of sorts still. It's often the case that when there's an initial implementation and specification, and maybe some tests, there's still many things found by the second implementation. Not entirely sure how to address that though, but we need some kind of incentive for folks to perform their due diligence.

**Comment by annevk** [comment]
Since my second comment got hidden for unclear reasons:

> I think we do have a quality problem of sorts still. It's often the case that when there's an initial implementation and specification, and maybe some tests, there's still many things found by the second implementation. Not entirely sure how to address that though, but we need some kind of incentive for folks to perform their due diligence.

File: lifecycle.md (hunk 1 above)
Comment (foolip) [reply lifecycle.md]: OK, so here tests came before the spec text even. I'll update.

Comment (foolip) [reply lifecycle.md]: Oh, actually it looks like it happened at the same time, per the now normal WHATWG working mode, and were merged together.

File: lifecycle.md (hunk 2 above)
Comment (foolip) [reply lifecycle.md]: Yes, the temptation to assume tests are good and not think too critically about them is pretty strong. Not sure what to do about it, apart from sketching what I think good behavior is under "Second implementation"

File: lifecycle.md (hunk 1 above)
Comment (annevk) [reply lifecycle.md]: Right, but I think that's rather critical as they end up influencing each other.

Comment (annevk) [reply lifecycle.md]: In particular I want someone reading this to realize you end up writing a better specification.

Comment (foolip) [reply lifecycle.md]: I guess that interplay happened for Fetch as well? Maybe I can say "Here and for the above case, the process of writing tests influenced the spec text and made it better."? Or is it important to also spell out the influence of spec text on tests, is that different when they happen together?

**Comment by foolip** [comment]
I think I'll change the order of examples (to not be alphabetical) to make the points clearer.

**Pull request merged**